enhancements/microshift/multus-cni-for-microshift.md

```diff
@@ -0,0 +1,519 @@
+* As a MicroShift admin, I want to add additional interfaces to certain Pods so that I can
+  start slowly migrating an existing solution to MicroShift.
+
+### Goals
+
+- Provide optional Multus CNI for MicroShift clusters that can be added to existing clusters
+- Provide container network plugins that are planned to be supported with MicroShift and Multus,
+  meaning:
+  - IPAMs: host-local, dhcp, static
+  - CNIs:
+    - bridge - must have
+    - macvlan, ipvlan - stretch goal
+
+### Non-Goals
+
+- Automatically removing Multus from the cluster upon RPM uninstall
+- Support Multus for multi-node deployments of MicroShift
+- Providing network policies, support for Services via CNIs other than ovn-kubernetes,
+  or admission webhook for Multus
+
+## Proposal
+
+Deliver Multus for MicroShift as an optional RPM containing required manifests that will be applied
+during MicroShift's start. There should be little to no changes to MicroShift itself as we want
+Multus CNI for MicroShift to fit optional components pattern.
+
+Manifests for deploying Multus on MicroShift will be based on existing manifests for OpenShift,
+but they will differ because OpenShift uses thick architecture Multus whereas MicroShift will use
+thin architecture Multus.
+
+How should cleanup of the Multus artifacts look like is an open question (see Open Questions section).
+
+### Workflow Description
+
+**User** is a human user responsible for setting up and managing Edge Devices.
+**Application** is user's workload that intends to use additional interfaces.
+
+#### Installation and usage on RHEL For Edge (ostree)
+
+> In this workflow, it doesn't matter if the device is already running R4E with existing MicroShift cluster.
+> Deployment of new commit requires reboot which will force recreation of the Pod networking after adding Multus.
+
+1. User gathers all information about the networking environment of the edge device.
+1. User prepares NetworkAttachmentDefinition (NAD) CRs that will allow Application to be part of the specified network.
+1. User prepares ostree commit that contains:
+   - (optional) Init procedures to configure OS for usage of the additional network
+   - MicroShift RPMs
+   - Multus for MicroShift RPM
+   - NetworkAttachmentDefinition CRs
+   - Application using mentioned NetworkAttachmentDefinition CRs
+1. User deploys the ostree commit onto the edge device.
+1. Edge device boots:
+1. (optional) Init procedures are configuring OS and networks
+1. MicroShift starts
+1. MicroShift applies Multus' manifests
+1. MicroShift applies Application's manifests that include NetworkAttachmentDefinitions
+1. Application's Pod are created, Multus inspects Pod's annotations and sets up 
+   additional interfaces based on matching NetworkAttachmentDefinitions
+1. Application's containers are running, they can utilize additional interfaces
+
+#### Installation and usage on RHEL (rpm)
+
+##### Adding to existing MicroShift cluster
```

Comment:
We could add this to the existing RPM test scenario.

Reply:
Great idea, added this to testing plan